packages/driver/cypress/e2e/commands/navigation.cy.js

```diff
@@ -2540,104 +2595,204 @@ describe('src/cy/commands/navigation', () => {
         })
       })
 
-      // https://github.com/cypress-io/cypress/issues/19230
-      it('filters page load events when going back with window navigation', () => {
-        const emit = cy.spy(Cypress, 'emit').log(false).withArgs('navigation:changed')
-
-        cy
-        .visit('/fixtures/generic.html')
-        .get('#hashchange').click()
-        .window().then((win) => {
-          return new Promise((resolve) => {
-            cy.once('navigation:changed', resolve)
+      describe('filters page load events when going back with window navigation when testIsolation=legacy', { testIsolation: 'legacy' }, () => {
+        // https://github.com/cypress-io/cypress/issues/19230
+        it('when going back with window navigation', () => {
+          const emit = cy.spy(Cypress, 'emit').log(false).withArgs('navigation:changed')
 
-            win.history.back()
-          }).then(() => {
+          cy
+          .visit('/fixtures/generic.html')
+          .get('#hashchange').click()
+          .window().then((win) => {
             return new Promise((resolve) => {
               cy.once('navigation:changed', resolve)
 
-              win.history.forward()
-            })
-          })
-        })
+              win.history.back()
+            }).then(() => {
+              return new Promise((resolve) => {
+                cy.once('navigation:changed', resolve)
 
-        cy.get('#dimensions').click()
-        .window().then((win) => {
-          return new Promise((resolve) => {
-            cy.on('navigation:changed', (event) => {
-              if (event.includes('(load)')) {
-                resolve()
-              }
+                win.history.forward()
+              })
             })
-
-            win.history.back()
           })
-          .then(() => {
+
+          cy.get('#dimensions').click()
+          .window().then((win) => {
             return new Promise((resolve) => {
-              cy.on('navigation:changed', resolve)
+              cy.on('navigation:changed', (event) => {
+                if (event.includes('(load)')) {
+                  resolve()
+                }
+              })
+
               win.history.back()
             })
+            .then(() => {
+              return new Promise((resolve) => {
+                cy.on('navigation:changed', resolve)
+                win.history.back()
+              })
+            })
+            .then(() => {
+              expect(emit.firstCall).to.be.calledWith(
+                'navigation:changed',
+                'page navigation event (load)',
+              )
+
+              expect(emit.secondCall).to.be.calledWith(
+                'navigation:changed',
+                'page navigation event (before:load)',
+              )
+
+              expect(emit.thirdCall).to.be.calledWith(
```

Comment:
nit: I think it reads a bit more consistently with `emit.getCall(2)` than `emit.thirdCall`

Reply:
done! [`8d0142f` (#23503)](https://github.com/cypress-io/cypress/pull/23503/commits/8d0142f968ac4e143a789023f26334e99929e06d)